csharp/autobuilder/Semmle.Autobuild/MsBuildRule.cs

```diff
@@ -57,7 +57,14 @@ public BuildScript Analyse(Autobuilder builder, bool auto)
                 var command = new CommandBuilder(builder.Actions);
 
                 if (vsTools != null)
+                {
                     command.CallBatFile(vsTools.Path);
+                    // `vcvarsall.bat` sets a default Platform environment variable,
+                    // which may not be compatible with the supported platforms of the
+                    // given project/solution. Unsetting it means that the default platform
+                    // of the project/solution is used instead.
+                    command.RunCommand("set Platform=&& type NUL", quoteExe: false);
```

Comment:
(a) doesn't work, because we need to _unset_ the environment variable that is set later by `vcvarsall.bat`. (b) might work, I'll try it out.

Reply:
(b) did not work (https://github.com/hvitved/ql/commit/15695069618db9e4842fb9ccb1123a0cdb44d699, test job: https://jenkins.internal.semmle.com/job/LGTM/job/Autobuild-Tests-v2/job/CSharp/20/console). It appears that appending `/p:Platform= ` sets `Platform` to the empty string, rather than removing the property.

Reply:
Do you think something like this could work instead?
```
            var set = new CommandBuilder(builder.Actions);
            set.RunCommand("SET Platform=");
            ret &= set.Script;
```

Reply:
No, that would result in a command like
```
cmd.exe /C CALL ^"C:\Program Files ^(x86^)\Microsoft Visual Studio\2017\Professional\VC\Auxiliary\Build\vcvars32.bat^" && <build command> && SET Platform=
```
while what we need is
```
cmd.exe /C CALL ^"C:\Program Files ^(x86^)\Microsoft Visual Studio\2017\Professional\VC\Auxiliary\Build\vcvars32.bat^" && SET Platform=&& <build command>
```

Reply:
Ah sorry. The thing I was suspicious of was that we should be using `builder.NextCommand()` somehow, for example
```
    command.CallBatFile(vsTools.Path);
    command.SetEnvironment("Platform", "");
```
Then in `class CommandBuilder`,
```
        public CommandBuilder SetEnvironment(string env, string value)
        {
            NextCommand();
            arguments.Append(" SET ").Append(env).Append("=").Append(value);
            return this;
        }
```

Reply:
That would result in
```
cmd.exe /C CALL ^"C:\Program Files ^(x86^)\Microsoft Visual Studio\2017\Professional\VC\Auxiliary\Build\vcvars32.bat^" && SET Platform= && <build command>
```
which does not unset `Platform` (note the space between `Platform=` and `&&`). This is also why I invoke `set Platform=&& type NUL` rather than just `set Platform=`.